packages/devtools_app/test/test_infra/scenes/standalone_ui/editor_service/editor_server.dart

```diff
@@ -74,3 +86,204 @@ abstract class EditorServer {
   /// `debugSessionStopped` event.
   void sendDebugSessionStopped(EditorDebugSession debugSession);
 }
+
+/// A wrapper over DTD for providing editor functionality.
+///
+/// This class is useful for tests and the "mock editor" but could in theory
+/// also be used by an editor directly (if it was written in Dart, and had
+/// access to this class).
+///
+/// Since this class is intended to represent what real IDEs may do, any changes
+/// made here to match changes made to [EditorClient] should be considered
+/// carefully to ensure they are not breaking changes to already-shipped
+/// editors.
+abstract class DtdEditorServer extends EditorServer {
+  // TODO(dantup): Once the postMessage code is gone, merge DtdEditorServer and
+  //  EditorServer to simplify things.
+  DtdEditorServer(this._dtdUri) {
+    // Connect editor automatically at launch.
+    unawaited(connectEditor());
+  }
+
+  final Uri _dtdUri;
```

Comment:
It's possible - we might want to let the simulated editor set some workspace roots for extensions (I think currently we just mock them). Any objection to changing the `uri` from `String` to `Uri` in:

```dart
typedef DTDConnectionInfo = ({String? uri, String? secret});
```

Many of these places are already using `Uri`s and I feel like it might be better to change the other places to use `Uri`s than these to `String`s.

Reply:
I believe this would be a breaking change to devtools_shared, so would prefer that we leave it as is for now and we can add a TODO to consider doing this later.

Reply:
Ah yes, ofc.

I realised when I went back to this that getting a secret to here is going to be complicated because for now we're just spawning the DTD instance as a VS Code task - we don't have a way to get the secret over here. I'll need to think a bit more about this so will leave for another PR.